<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Robot name parameter -->
    <arg name="robot_name" default="locobot" />
    
    <!-- Configuration file path -->
    <arg name="poses_config" default="$(find object_scout)/config/poses.yaml" />
    <arg name="max_objects" default="3" />

    <!-- Topic parameters -->
    <arg name="object_marker_topic" default="/$(arg robot_name)/object_markers" />
    <arg name="bbox_depth_topic" default="/$(arg robot_name)/camera/yolo/bbox_depth" />
    
    <!-- Visualization parameters -->
    <arg name="rviz_frame" default="map" />
    <arg name="rvizconfig" default="$(find yolo_vision)/rviz/xslocobot_description_yolo.rviz" />
    <arg name="use_rviz" default="true" doc="Set to false to disable RViz visualization" />


    
    <!-- Include the YOLO vision system -->
    <include file="$(find yolo_vision)/launch/object_mapper.launch">
        <arg name="robot_name" value="$(arg robot_name)"/>
        <arg name="rviz_frame" value="$(arg rviz_frame)"/>
        <arg name="rvizconfig" value="$(arg rvizconfig)"/>
        <arg name="object_marker_topic" value="$(arg object_marker_topic)"/>
        <arg name="bbox_depth_topic" value="$(arg bbox_depth_topic)"/>
        <arg name="use_rviz" value="false"/>
    </include>
    
    <!-- Launch the coordinator node with all components packaged inside -->
    <node name="scout_coordinator" pkg="object_scout" type="scout_coordinator_remote_node.py" output="screen">
        <param name="robot_name" value="$(arg robot_name)" />
        <param name="poses_config" value="$(arg poses_config)" />
        <param name="object_marker_topic" value="$(arg object_marker_topic)" />
        <param name="bbox_depth_topic" value="$(arg bbox_depth_topic)" />
        <param name="max_objects" value="$(arg max_objects)" />
    </node>
    
    <!-- RViz (conditionally launched) -->
    <group if="$(arg use_rviz)">
        <node name="rviz" pkg="rviz" type="rviz" 
                args="-f $(arg rviz_frame) -d $(arg rvizconfig)"
                ns="$(arg robot_name)">
            <remap from="/clicked_point" to="clicked_point"/>
            <remap from="/initialpose" to="initialpose"/>
            <remap from="/move_base_simple/goal" to="move_base_simple/goal"/>
        </node>
    </group>
</launch>